runtime/src/runtime.c

```diff
@@ -29,19 +29,12 @@ int lf_retrieve(struct _lf_device *device, struct _fmr_result *result) {
 	return lf_error;
 }
 
-lf_return_t lf_invoke(struct _lf_module *module, lf_function function, lf_type ret, struct _lf_ll *parameters) {
-	lf_assert(module, failure, E_NULL, "No module was specified for function invocation.");
-
-	/* If the module has no device, assume the invocation is for the current device. */
-	if (!module->device) {
-		lf_debug("No device specified for module %s, using current device.", module->name);
-		module->device = lf_get_current_device();
-	}
-
-	lf_assert(module->device, failure, E_NO_DEVICE, "The module '%s' has no target device. Did you attach?", module->name);
+lf_return_t lf_invoke(struct _lf_device *device, struct _lf_module *module, lf_function function, lf_type ret, struct _lf_ll *parameters) {
+	lf_assert(device, failure, E_NULL, "No device was provided to '%s'.", __PRETTY_FUNCTION__);
```

Comment:
Is `__PRETTY_FUNCION__` a C builtin?

Reply:
Compiler construct for the function name string